dsgrid/dimension/base_models.py

```diff
@@ -120,4 +132,37 @@ def check_required_dimensions(dimensions, tag):
     missing = required_dim_types.difference(dimension_types)
     if missing:
         raise ValueError(f"Required dimension(s) {missing} are not in {tag}.")
+
     check_uniqueness((x.dimension_type for x in dimensions), tag)
+
+
+def check_timezone_in_base_geography(dim_config):
+    """Check that a project's base geography dimension contains valid timezones
+    in records.
+
+    Parameters
+    ----------
+    dimension : DimensionType
+
+    Raises
+    ------
+    DSGInvalidDimension
+        Raised if a required dimension is not provided.
+
+    """
+    dimension = dim_config.model
+    if dimension.dimension_type != DimensionType.GEOGRAPHY:
+        raise ValueError(
+            f"Dimension has type {dimension.dimension_type}, "
+            "Can only check timezone for Geography."
+        )
+
+    tz = set(TimeZone) - TimeZone.NONE
+    record_tz = {rec.time_zone for rec in dimension.records}
+
+    diff = record_tz.difference(tz)
+    if diff:
+        raise DSGInvalidDimension(
```

Comment:
Should this exception be consistent with the one on line 155? It' not called as part of Pydantic validation, right? (so then it can be any exception type)